pandas/tests/test_join.py

```diff
@@ -348,3 +348,20 @@ def test_merge_join_categorical_multiindex():
     result = a.join(b, on=["Cat1", "Int1"])
     expected = expected.drop(["Cat", "Int"], axis=1)
     assert_frame_equal(expected, result)
+
+
+def test_left_index_and_right_index_true():
+    # From issue 28189
+
+    pdf = DataFrame(
+        range(4), columns=["value"], index=Index(Categorical(["1"] * 4), name="idx")
+    )
+    agg = pdf.groupby("idx").agg(np.sum)["value"]
+
+    result = merge(pdf, agg, how="left", left_index=True, right_index=True)
+    result = result.reset_index(drop=True)
```

Comment:
Is the reset_index required or can the index be included as part of `expected`?

Reply:
I left it out of the test because of the issues @hugoecarl mentioned on the PR. If the index are included, the test will break once they're fixed, but if you want, it could be included and I would need to change the `expected` df.

Reply:
What do you mean by "the test will break once they're fixed"?

Reply:
What @guipleite is trying to say is that when you use the left-join, there is a problem with the index values output. The bug is related with the issues @hugoecarl pointed here in the PR. With that been said, if we include index as a part of the expected df, we would have to force the wrong output to prove that the bug pointed on this PR is solved, then when those other issues are corrected, this test would be outdated. Does it make sense to you?

Reply:
IIUC you are saying there are multiple bugs and this solves one of them, but the other would ned to be solved separately right? If so is there an open issue for that? Might have missed it in this PR

Reply:
Yes! This is particularly solving issue #28189 . This other bug that we accidentally found have already been reported in issues #28220 and #28243 .

Reply:
Hmm but don't those issues have to do with missing data on merge? Maybe missing the point but don't see how applicable here. Isn't the expected index still just `Index(Categorical(["1"] * 4), name="idx")`?

Reply:
Exactly that! We just added the support to int8 and int16 type that uncovered this output bug.